src/main/java/edu/cornell/kfs/sys/service/impl/ISOFIPSConversionServiceImpl.java

```diff
@@ -68,87 +76,135 @@ public class ISOFIPSConversionServiceImpl implements ISOFIPSConversionService {
     
     private static final Logger LOG = LogManager.getLogger(ISOFIPSConversionServiceImpl.class);
     
-    private static final String NO_ISO_TO_FIPS_MAPPINGS_ERROR_MESSAGE = "No ISO-to-FIPS Country generic mapping found for ISO Country code : {0}";
-    private static final String MANY_ISO_TO_FIPS_MAPPINGS_ERROR_MESSAGE = "More than one ISO-to-FIPS Country generic mapping found for ISO Country code : {0}";
-    private static final String ONE_TO_ONE_ISO_TO_FIPS_MAPPING_MESSAGE = "One ISO-to-FIPS Country generic mapping found: ISO Country code : {0} mapped to FIPS Country code : {1}";
-    private static final String NO_FIPS_TO_ISO_MAPPINGS_ERROR_MESSAGE = "No FIPS-to-ISO Country generic mapping found for FIPS Country code : {0}";
-    private static final String MANY_FIPS_TO_ISO_MAPPINGS_ERROR_MESSAGE = "More than one FIPS-to-ISO Country generic mapping found for FIPS Country code : {0}";
-    private static final String ONE_TO_ONE_FIPS_TO_ISO_MAPPING_MESSAGE = "One FIPS-to-ISO Country generic mapping found FIPS Country code : {0} mapped to ISO Country code : {1}";
-
-    private ISOFIPSCountryMapDao isoFipsCountryMapDao;
+    private ISOCountryService isoCountryService;
+    private CountryService countryService;
+    private ISOFIPSCountryMapService isoFipsCountryMapService;
+    private ConfigurationService configurationService;
     
     /**
-     * Return the single ACTIVE FIPS country code mapped to the ISO country code input parameter;
-     * Otherwise throw a NoISOtoFIPSMappingException or ManyISOtoFIPSMappingException.
+     * Return the single ACTIVE FIPS country code mapped to the ISO country code input parameter when the
+     * ISO Country is Active, the FIPS Country is Active, and a single Active ISO-FIPS Country Mapping exists;
      *
-     * Active status of the MAPPING IS USED for this data retrieval.
-     */
+     * NOTE: Active status IS USED for all three items in this data retrieval
+     */ 
     @Override
     public String convertISOCountryCodeToActiveFIPSCountryCode(String isoCountryCode) {
-        List<ISOFIPSCountryMap> mappingsFound = findManyActiveFIPSCountryCodesForISOCode(isoCountryCode);
-
+        if (getIsoCountryService().isISOCountryInactive(isoCountryCode)) {
+            LOG.error("convertISOCountryCodeToActiveFIPSCountryCode: " + MessageFormat.format(getConfigurationService().getPropertyValueAsString(CUKFSKeyConstants.ERROR_NO_ISO_TO_FIPS_MAPPINGS), isoCountryCode));
+            throw new NoISOtoFIPSMappingException(MessageFormat.format(getConfigurationService().getPropertyValueAsString(CUKFSKeyConstants.ERROR_NO_ISO_TO_FIPS_MAPPINGS), isoCountryCode));
+        }
+        
+        List<ISOFIPSCountryMap> mappingsFound = findManyActiveFIPSCountryMapsForISOCode(isoCountryCode);
         if (mappingsFound.isEmpty()) {
```

Comment:
instead of doing these .isEmpty checks you could use CollectionUtils, that would give you a null check for free